ms2deepscore/MS2DeepScore.py

```diff
@@ -0,0 +1,125 @@
+from typing import List
+from typing import Union
+import numpy as np
+from matchms.similarity.BaseSimilarity import BaseSimilarity
+from tqdm import tqdm
+
+from ms2deepscore.vector_operations import cosine_similarity
+from ms2deepscore.vector_operations import cosine_similarity_matrix
+from ms2deepscore import BinnedSpectrum
+from ms2deepscore import SpectrumBinner
+from ms2deepscore.models import SiameseModel
+
+
+class MS2DeepScore(BaseSimilarity):
+    """Calculate MS2DeepScore similarity scores between a reference and a query.
+
+    Using a trained model, binned spectrums will be converted into spectrum
+    vectors using a deep neural network. The MS2DeepScore similarity is then
+    the cosine similarity score between two spectrum vectors.
+
+    Example code to calcualte MS2DeepScore similarities between query and reference
+    spectrums:
+
+    .. code-block:: python
+
+        from matchms import calculate_scores()
+        from matchms.importing import load_from_json
+        from ms2deepscore import MS2DeepScore
+        from ms2deepscore.models import load_ms2ds_model
+
+        references = load_from_json("abc.json")
+        queries = load_from_json("xyz.json")
+
+        model = ... TODO: implement such a function
+        binned_references = model.spectrum_binner.transform(references)
+        binned_queries = model.spectrum_binner.transform(queries)
+
+        similarity_measure = MS2DeepScore(model)
+        scores = calculate_scores(binned_references, binned_queries)
+
+
+    """
+    def __init__(self, model, progress_bar: bool = False):
+        """
+
+        Parameters
+        ----------
+        model:
+            Expected input is a SiameseModel that has been trained on
+            the desired set of spectra.
+        intensity_weighting_power:
+            Spectrum vectors are a weighted sum of the word vectors. The given
+            word intensities will be raised to the given power.
+            The default is 0, which means that no weighing will be done.
+        progress_bar:
+            Set to True to monitor the embedding creating with a progress bar.
+            Default is False.
+        """
+        self.model = model
+        self.input_vector_dim = self.model.base.input_shape[1]  # TODO: later maybe also check against SpectrumBinner
+        self.output_vector_dim = self.model.base.output_shape[1]
+        self.disable_progress_bar = not progress_bar
+
+    def _create_input_vectors(self, binned_spectrum: BinnedSpectrum):
+        """Creates input vector for model.base based on binned peaks and intensities"""
+        X = np.zeros((1, self.input_vector_dim))
+
+        idx = np.array([int(x) for x in binned_spectrum.binned_peaks.keys()])
+        values = np.array([x for x in binned_spectrum.binned_peaks.values()])
+        X[0, idx] = values
+        return X
+
+    def pair(self, reference: BinnedSpectrum, query: BinnedSpectrum) -> float:
```

Comment:
I propose to input `Spectrum` instead of `BinnedSpectrum` here. The spectrums are then binned within this class. This prevents an important caveat that this code currently has, namely that a user might get a `SiameseModel` from somewhere, and a `SpectrumBinner` from somewhere else. But they might not be compatible with each other. 

Also [matchms](https://github.com/matchms/matchms/blob/master/matchms/similarity/BaseSimilarity.py) library takes `Spectrum` as input and not `BinnedSpectrum`.

See further discussion in: #23

Reply:
I updated the discussion in #23. So I will not add the `SpectrumBinner` here, but using `Spectrum` inputs instead of `BinnedSpectrum` could still be an interesting option.

Reply:
Actually, I just run a quick performance check: for the 76 spectra used in `test_user_workflow` binning the spectra takes 200x less time than the actual score computation. Based on that I will move the conversion step into the actual scoring method so that users won't have to deal with the conversion at all!

Reply:
Done. `MS2DeepScore` now uses `Spectrum` inputs.

Reply:
Cool!